functions.py

```diff
@@ -13,44 +14,64 @@ def search(path, primary_key):
 
 def creates_dir(schema):
     x = json.load(open(schema, 'r'))
-    p = os.path.join(os.getcwd(), x['database_name'])
-    os.mkdir(p)
-    for i in x['Tables']:
-        d = os.path.join(p, i['name'])
-        os.mkdir(d)
+    p = os.path.join(os.getcwd(), x[keys().database])
+    exists = True
+    if not os.path.exists(p):
+        exists = False
+        os.mkdir(p)
+        for i in x[keys().Tables]:
+            d = os.path.join(p, i[keys().name])
+            os.mkdir(d)
+    else:
+        for i in x[keys().Tables]:
+            d = os.path.join(p, i[keys().name])
+            if not os.path.exists(d):
+                exists = False
+                os.mkdir(d)
+    return exists
 
 
 def creates(schema, table, primary_key):
-    x = json.load(open(schema, 'r'))
-    path = os.getcwd() + '\\' + x['database_name'] + '\\' + table + '\\' + primary_key
-    f = open(path, 'w')
-    f.write("{\n")
-    i = 0
-    for k in x['Tables']:
-        if k['name'] != table:
-            i = i + 1
-        else:
-            break
-    x = x['Tables'][i]
-    i = len(x['columns'])
-    j = 0
-    for t in x['columns']:
-        f.write("\t\"" + t + "\" : \"0\"")
-        j = j + 1
-        if j < i:
-            f.write(",\n")
-    f.write("\n}\n")
+    f = open(schema, 'r')
+    x = json.load(f)
     f.close()
+    path = os.path.join(os.getcwd(), x[keys().database])
+    if not os.path.exists(path):
+        return False
+    path = path + '\\' + table + '\\' + primary_key
+    if not os.path.exists(path):
+        f = open(path, 'w')
+        f.write("{\n")

```

Comment:
What are you trying to do here ?
You can convert the object to json string. Google it.

you don’t need to serialize the object manually